go/runtime/client/host.go

```diff
@@ -0,0 +1,179 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/oasisprotocol/oasis-core/go/common/logging"
+	consensus "github.com/oasisprotocol/oasis-core/go/consensus/api"
+	"github.com/oasisprotocol/oasis-core/go/runtime/host"
+	"github.com/oasisprotocol/oasis-core/go/runtime/host/protocol"
+	runtimeRegistry "github.com/oasisprotocol/oasis-core/go/runtime/registry"
+	storage "github.com/oasisprotocol/oasis-core/go/storage/api"
+	"github.com/oasisprotocol/oasis-core/go/storage/mkvs/syncer"
+)
+
+var (
+	errMethodNotSupported   = errors.New("method not supported")
+	errEndpointNotSupported = errors.New("endpoint not supported")
+)
+
+type clientHost struct {
+	*runtimeRegistry.RuntimeHostNode
+
+	runtime   runtimeRegistry.Runtime
+	consensus consensus.Backend
+
+	stopCh chan struct{}
+	quitCh chan struct{}
+
+	logger *logging.Logger
+}
+
+// Start starts the client host.
+func (h *clientHost) Start() error {
+	go h.worker()
+	return nil
+}
+
+// Stop asks the client host to stop.
+func (h *clientHost) Stop() {
+	close(h.stopCh)
+}
+
+// Quit returns the channel which will signal when the client host has quit.
+func (h *clientHost) Quit() <-chan struct{} {
+	return h.quitCh
+}
+
+// Implements runtimeRegistry.RuntimeHostHandlerFactory.
+func (h *clientHost) GetRuntime() runtimeRegistry.Runtime {
+	return h.runtime
+}
+
+// Implements runtimeRegistry.RuntimeHostHandlerFactory.
+func (h *clientHost) NewRuntimeHostHandler() protocol.Handler {
+	return h
+}
+
+// Implements runtimeRegistry.RuntimeHostHandlerFactory.
+func (h *clientHost) NewNotifier(ctx context.Context, host host.Runtime) protocol.Notifier {
+	return nil
+}
+
+// Implements protocol.Handler.
+func (h *clientHost) Handle(ctx context.Context, body *protocol.Body) (*protocol.Body, error) {
```

Comment:
how does this end up getting used? I don't have any experience with RuntimeHostHandlerFactory.

Reply:
This gets called whenever the runtime calls a method on the host. On the client node we only support storage queries.